packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Providers/RestClientProvider.cs

```diff
@@ -345,6 +346,61 @@ private static void GetParamInfo(Dictionary<string, ParameterProvider> paramMap,
             }
         }
 
+        private static IReadOnlyList<ParameterProvider> BuildOrderedSpreadParameters(InputModelType inputModel)
+        {
+            // create the model provider
+            if (ClientModelPlugin.Instance.TypeFactory.CreateModel(inputModel) is not ModelProvider modelProvider)
+                throw new InvalidOperationException($"Unable to create model for input model {inputModel.Name}.");
+
+            var fullConstructor = modelProvider.FullConstructor;
+            var primaryConstructor = modelProvider.Constructors.FirstOrDefault(c => c != fullConstructor);
```

Comment:
I am not sure how we are getting a deterministic constructor.  Are we looking for the smallest param set public ctor?

Reply:
The updated logic now searches for the full constructor by checking the parameter count + the method signature modifier.